### ISSUE TITLE
make test-unit -> db_text_search

### ISSUE THREAD

**Issue by eckelj** [planetmint/planetmint#92]
asset and metadata come in type : any, this cannot be indexed. 

we therefore have to change this into string and convert this back to a json structure before putting it into any other objects.

	 be aware: that there is a difference in running : 
	 	- 'make test' and to just run 'pytest'
	 	- execute 'docker-compose build planetmint' piror to run 'make test'
	 	- you can execute 'pip/pipenv install -e ".[dev]"' to get a similar environment to make test

**Comment by eckelj** [comment]
we need to enable text-search at
enable tarantool:text_serach which is currently commented out,

the following tests need to be passing again:
* define a tarantool:test-queries.py test_serach example, 
* test_bigchaindb_api.py: test_text_search
* tests/web/test_assests.py
* tests/web/test_metadata.py

**Comment by eckelj** [comment]
the following PR contains some changes that enable a simple text search on tarantool:
https://github.com/liviu-lesan/planetmint/pull/6/files

however, some specific type and conversion issues have to be tackled so that the tests pass again.

**Comment by LaurentDeMontBlanc** [comment]
The tarantool documentation offers a tutorial for an indexed pattern search that offers both speed and flexibility. I think the best approach would be to define the functions in tarantool and call them with the python connector.

ToDos:
 - setup `indexed_pattern_search` function in tarantool
 - add secondary indices for `asset` and `metadata` to make them searchable with `indexed_pattern_seach`
 - use `space.call()` method to call `indexed_pattern_search` from the python connector

**Comment by LaurentDeMontBlanc** [comment]
I tried to use the `indexed_pattern_search` for `assets` like so:

Prerequisities:
 - copy the `indexed_pattern_search` from [here](https://www.tarantool.io/en/doc/latest/tutorials/lua_tutorials/#indexed-pattern-search) and paste it in the tarantool cli

1.) Creating space, schema and indices
```
unix/:/var/run/tarantool/tarantool.sock> s = box.schema.space.create("assets")
---
...
unix/:/var/run/tarantool/tarantool.sock> s:format({{name='data' , type='string'}, {name='tx_id', type='string'}, {name='asset_id', type='string'}})
---
...


unix/:/var/run/tarantool/tarantool.sock> s:create_index('primary', {})
---
- unique: true
  parts:
  - type: string
    is_nullable: false
    fieldno: 1
  hint: true
  id: 0
  type: TREE
  space_id: 512
  name: primary
...

unix/:/var/run/tarantool/tarantool.sock> s:create_index('secondary', {unique=false,parts={1,'string'}})
---
- unique: false
  parts:
  - type: string
    is_nullable: false
    fieldno: 1
  hint: true
  id: 1
  type: TREE
  space_id: 512
  name: secondary
...
```
2.) running `indexed_pattern_search`
```
unix/:/var/run/tarantool/tarantool.sock> indexed_pattern_search("assets", 1, "key:")
---
- - ['key:valuestring', 'tx_id', 'asset_id']
...
```
However that worked only for the start of the string as you can see in the example below:
```
unix/:/var/run/tarantool/tarantool.sock> indexed_pattern_search("assets", 1, "val")
---
- []
...
```
Taken from the NOTE#2 states that lua has some "magic characters" for matching patterns. Investigating if similar to regex

**Comment by eckelj** [comment]
@roninx991 : @LaurentDeMontBlanc mentioned to use lua based regex: is this 
https://www.lua.org/pil/20.1.html   http://www.lua.org/manual/5.2/manual.html#6.4.1
so maybe LUA can take over the tasks of step#2 and #3 
 #2 und #3 von indexed_pattern_search passiert (edited)

**Comment by andreitricolici** [comment]
In this example it is using . (dot) to do a text search. try to run your query like this.
**indexed_pattern_search("assets", 1, ".val.")**
![DeepinScreenshot_select-area_20220610111434.png](https://images.zenhubusercontent.com/62a089a50e08612d1d3d9ca6/e4d37a5c-dfea-47b1-9427-b96d3248e295)

**Comment by LaurentDeMontBlanc** [comment]
@eckelj I've removed the limitation for now form the `indexed_pattern_search` and added the `text_search` to `planetmint/backend/tarantool/query.py`. It adds `.` on both ends of the string to match for everything at the moment. 

The current limitation is that it only works with strings. I therefor had to change the `planetmint/backend/tarantool/schema.py` so that `meta_data` and `asset` data is stored as JSON string. I adjusted all tarantool specific methods that read/write `meta_data` or `assets` to the database.

An issue that I encountered is that `get_asset` does not return the whole `asset` but only the `data` property of said asset. `get_assets` on the other hand calls `get_asset` for every `asset_id` that is passed to it and expects an `id` property on the `_returned_data`. I will investigate which parts are affected by this.
